src/components/SiteTrackingRequired/SiteTrackingRequired.test.js

```diff
@@ -0,0 +1,110 @@
+
+describe('site tracking', () => {
+  afterEach(cleanup);
+
+  it('should show free account messages', () => {
+    // Arrange
+    const reason = SiteTrackingNotAvailableReasons.freeAccount;
+    const dependencies = {
+      dopplerLegacyClient: {
+        activateSiteTrackingTrial: async () => false,
+      },
+    };
+
+    // Act
+    const { getByText } = render(
+      <AppServicesProvider forcedServices={dependencies}>
+        <DopplerIntlProvider>
+          <SiteTrackingRequired reason={reason} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    expect(getByText('reports.upgrade_account_free_title')).toBeInTheDocument();
+  });
+
+  it('should show trial not accepted messages', () => {
+    // Arrange
+    const reason = SiteTrackingNotAvailableReasons.trialNotAccepted;
+    const dependencies = {
+      dopplerLegacyClient: {
+        activateSiteTrackingTrial: async () => false,
+      },
+    };
+
+    // Act
+    const { getByText } = render(
+      <AppServicesProvider forcedServices={dependencies}>
+        <DopplerIntlProvider>
+          <SiteTrackingRequired reason={reason} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    expect(getByText('reports.allow_enable_trial_title')).toBeInTheDocument();
+  });
+
+  it('should show not domains messages', () => {
+    // Arrange
+    const reason = SiteTrackingNotAvailableReasons.thereAreNotDomains;
+    const dependencies = {
+      dopplerLegacyClient: {
+        activateSiteTrackingTrial: async () => false,
+      },
+    };
+
+    // Act
+    const { getByText } = render(
+      <AppServicesProvider forcedServices={dependencies}>
+        <DopplerIntlProvider>
+          <SiteTrackingRequired reason={reason} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    expect(getByText('reports.datahub_not_domains_title')).toBeInTheDocument();
+  });
+
+  it('should be activate the trial and redirect to control panel site tracking settings', async () => {
+    // Arrange
+    const reason = SiteTrackingNotAvailableReasons.trialNotAccepted;
+    const dependencies = {
+      dopplerLegacyClient: {
+        activateSiteTrackingTrial: async () => true,
+      },
+      appConfiguration: { dopplerLegacyUrl: 'localhost:3000' },
+      window: { location: { href: '/ControlPanel/CampaignsPreferences/SiteTrackingSettings' } },
+    };
+
+    // Act
+    const { getByText, container } = render(
+      <AppServicesProvider forcedServices={dependencies}>
+        <DopplerIntlProvider>
+          <SiteTrackingRequired reason={reason} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    expect(getByText('reports.allow_enable_trial_title')).toBeInTheDocument();
+    expect(dependencies.window.location.href).toEqual(
+      '/ControlPanel/CampaignsPreferences/SiteTrackingSettings',
+    );
```

Comment:
These are not the real assertion, are preliminary ones to verify if the arrange was fine.

Reply:
i removed